langchain4j-neo4j/src/main/java/dev/langchain4j/store/embedding/neo4j/Neo4jEmbeddingUtils.java

```diff
@@ -0,0 +1,83 @@
+package dev.langchain4j.store.embedding.neo4j;
+
+import dev.langchain4j.data.document.Metadata;
+import dev.langchain4j.data.embedding.Embedding;
+import dev.langchain4j.data.segment.TextSegment;
+import dev.langchain4j.store.embedding.EmbeddingMatch;
+import org.neo4j.driver.Record;
+import org.neo4j.driver.Value;
+import org.neo4j.driver.Values;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+public class Neo4jEmbeddingUtils {
+    
+    /* not-configurable strings, used in `UNWIND $rows ...` statement */
+    public static final String EMBEDDINGS_ROW_KEY = "embeddingRow";
+    public static final String ID_ROW_KEY = "id";
+    
+    /* default configs */
+    public static final String DEFAULT_EMBEDDING_PROP = "embedding";
+    public static final String PROPS = "props";
+    public static final String DEFAULT_IDX_NAME = "langchain-embedding-index";
+    public static final String DEFAULT_LABEL = "Document";
+    public static final String DEFAULT_TEXT_PROP = "text";
+    
+
+    public static EmbeddingMatch<TextSegment> toEmbeddingMatch(Neo4jEmbeddingStore store, Record neo4jRecord) {
+        var node = neo4jRecord.get("node").asNode();
```

Comment:
we should make configurable what we actually want to return from the database and then also from the find method, otherwise it might be a lot of data that is returned.